src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -607,14 +607,14 @@ public void PatternTests_UnixEscape(string pattern, string[] sourceFiles, string
         }
 
         [Fact]
-        [PlatformSpecific(TestPlatforms.Windows)]  // Long path segment in search pattern throws PathTooLongException
+        [PlatformSpecific(TestPlatforms.Windows)]  // Long path segment in search pattern throws DirectoryNotFoundException
         public void WindowsSearchPatternLongSegment()
         {
-            // Create a path segment longer than the normal max of 255
+            // Create a path segment longer than the extended max
             DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
-            string longName = new string('k', 257);
```

Comment:
Whatever it throws/doesn't throw we should test for the new behavior. There were two "PathTooLong" cases before- one was for overall path length, the other was for segment too long. This was testing the second case.